### DISCUSSION TITLE
How do I add extra packages through `extraPackages`?

### DISCUSSION THREAD

**Discussion by sylvesterroos** [shanesveller/nix-beam-flakes#13] [Q&A] [answered]
What am I doing wrong here?

```nix
devShell.extraPackages = [
  git
];
```

I get the following error:
```
error: undefined variable 'git'

       at /nix/store/lhr1v8nir1a7ab1xvbrhng2i0r6mvxph-source/flake.nix:28:13:

           27| 	  devShell.extraPackages = [
           28|             git
             |             ^
           29| 	  ];
```

I'm not sure where I should get the package definition from.

**Comment by shanesveller** [comment, marked as answer]
Hi there! There is no default source of packages that is automatically in scope, syntactically speaking, so you would need to provide that context in your own flake definition. This can be done via the surrounding syntax constructs, such as function arguments or a let-binding, for example.

You can see more information about the available arguments that `flake-parts` provides [here](https://flake.parts/module-arguments). I could potentially be more explicit in the README about this, but some amount of practical familiarity with `flake-parts` will be necessary to be successful with this library, and would potentially benefit your other Nix use anyway.

Adapting the [example flake](https://github.com/shanesveller/nix-beam-flakes/blob/ceaa76b13abf9ecdf488327c8bcb67517e981269/dev/example/flake.nix#L11-L34) to your question here, you can use the `flake-parts` [perSystem](https://flake.parts/options/flake-parts#opt-perSystem) argument to acquire a version of your `nixpkgs` input that already honors the system being defined. This means you can write `pkgs.something` and not `inputs.nixpkgs.legacyPackages.aarch64-darwin.something`:


```nix
outputs = inputs @ {
    beam-flakes,
    flake-parts,
    ...
  }:
    flake-parts.lib.mkFlake {inherit inputs;} {
      imports = [beam-flakes.flakeModule];

      systems = ["aarch64-darwin" "x86_64-darwin" "x86_64-linux"];

      perSystem = { pkgs, ... }: {
        beamWorkspace = {
          enable = true;
          # your other settings here ...
          devShell.extraPackages = [ pkgs.git ];
        };
      };
    };
```

**Reply by sylvesterroos** [reply]
Thanks a lot! 😄
